src/exoticatechnologies/modifications/exotics/impl/HullmodExotic.kt

```diff
@@ -29,24 +32,54 @@ open class HullmodExotic(
+            // <this exotic>, <parent FleetMember> + <list of expected variants> + <list of variants we installed on>
+            val mutableVariantList = moduleSlotList
+                    .map { slot -> member.variant.getModuleVariant(slot) }
+                    .toMutableList()
+            // Obviously, add the 'member' variant to the list as well
+            mutableVariantList.add(member.variant)
+            // Now forget about the mutable version and use a immutable version
+            val variantList = mutableVariantList.toList()
+
+            // Iterate through each slot, getting their variant and trying to install there
             for (slot in moduleSlotList) {
                 val moduleVariant = member.variant.getModuleVariant(slot)
                 logger.info("onInstall()\tslot: ${slot}\tmoduleVariant: ${moduleVariant}")
                 if (moduleVariant == null) continue
                 val mods = get(member, moduleVariant)
                 logger.info("onInstall()\tmods: ${mods}")
                 mods?.let { nonNullMods ->
-                    logger.info("onInstall()\t--> installWorkaround()\tmember: ${member}, moduleVariant: ${moduleVariant}, mods: ${mods}, exotic: ${this}")
-                    installWorkaround(member, moduleVariant, nonNullMods, this)
-                    logger.info("onInstall()\t--> installHullmodOnVariant()\tmoduleVariant: ${moduleVariant}")
-                    installHullmodOnVariant(moduleVariant)
+                    if (HullmodExoticHandler.shouldInstallHullmodExoticToVariant(
+                            hullmodExotic = this,
+                            parentFleetMember = member,
+                            variant = moduleVariant,
+                            variantList = Optional.of(variantList)
+                    )) {
+                        logger.info("onInstall()\t--> installWorkaround()\tmember: ${member}, moduleVariant: ${moduleVariant}, mods: ${mods}, exotic: ${this}")
+                        installWorkaround(member, moduleVariant, nonNullMods, this)
+                        logger.info("onInstall()\t--> installHullmodOnVariant()\tmoduleVariant: ${moduleVariant}")
+                        installHullmodOnVariant(moduleVariant)
+                        // And mark it as installed in the HullmodExoticHandler
+                        HullmodExoticHandler.installHullmodExoticToVariant(
+                                hullmodExotic = this,
+                                parentFleetMember = member,     //oh lets hope this one works out ...
+                                variant = moduleVariant
+                        )
+                    }
                 }
             }
         }
         installHullmodOnVariant(member.variant)
+        HullmodExoticHandler.installHullmodExoticToVariant(
+                hullmodExotic = this,
+                parentFleetMember = member,
+                variant = member.variant
+        )
```

Comment:
somehow, i believe this one should also check before installing.

however, the bad news is - this one differs from the installing on child modules like up there, due to lack of `installWorkaround(...)` - for which, i admit, i completely forgot why it's even a thing or exists.

I suppose we can try dropping the `installWorkaround` a bit later on.

but for the time being, lets try using `HullmodExoticHandler.installOnVariant(hullmodExotic, parentFleetMember, variant, useWorkaround)` for both of these cases

Reply:
as previously mentioned, we're merely using the flows here. the HullmodExoticHandler is doing everything. resolving.